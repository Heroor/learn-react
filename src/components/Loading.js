import React from 'react'
import PropTypes from 'prop-types'

const Loading = ({
  children,

}) => {
  return (
    <div className="loading-wrap">
      {children}
    </div>
  )
}

Loading.propTypes = {
  children: PropTypes.node
}

Loading.defaultProps = {
  children: 'Loading...'
}

export default Loading