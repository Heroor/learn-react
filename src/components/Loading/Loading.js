import React from 'react'
import PropTypes from 'prop-types'
import './index.css'

const Loading = ({
  children
}) => (
  <div className="loading-wrap">
    {children}
  </div>
)

Loading.propTypes = {
  children: PropTypes.node
}

Loading.defaultProps = {
  children: 'Loading...'
}

export default Loading
