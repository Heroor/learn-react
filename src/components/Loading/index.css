.loading-wrap img {
  width: 40px;
  height: 40px;
  font-size: 0;
  animation: rotate 2s infinite;
}

@keyframes rotate {
  to {
    transform: rotate(360deg);
  }
}