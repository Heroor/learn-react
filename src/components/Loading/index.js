import React from 'react'
import PropTypes from 'prop-types'

const Loading = ({
  children
}) => {
  return (
    <div className="loading-wrap">
      {children}
    </div>
  )
}

Loading.propTypes = {
  children: PropTypes.node
}

Loading.defaultProps = {
  children: 'Loading...'
}


// use the "with" prefix to name the HOC
export const withLoading = (Component) =>
  ({ isLoading, loadingChildren = null, ...rest }) =>
    isLoading ?
    <Loading>{loadingChildren}</Loading> :
    <Component {...rest}/>


export default Loading