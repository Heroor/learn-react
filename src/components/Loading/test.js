import React from 'react'
import ReactDOM from 'react-dom'
import Loading from './index'

import renderComponentToSnapshot from '../../config/TestConfig'

// Loading test
describe('Loading', () => {
  it('render without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Loading/>, div)
  })

  test('has a calid snapshot', () => {
    renderComponentToSnapshot(<Loading>is loading</Loading>)
  })
})