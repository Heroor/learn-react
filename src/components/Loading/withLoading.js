import React from 'react'
import Loading from './Loading'

// use the "with" prefix to name the HOC
const withLoading = (Component) =>
  ({ isLoading, loadingChildren = null, ...rest }) =>
    isLoading ?
    <Loading>{loadingChildren}</Loading> :
    <Component {...rest}/>

export default withLoading