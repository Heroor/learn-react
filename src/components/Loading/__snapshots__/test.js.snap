// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading has a calid snapshot 1`] = `
<div
  className="loading-wrap"
>
  is loading
</div>
`;
